tutorials/adding_new_layers.md

```diff
@@ -20,71 +31,140 @@
-- Open [filterbar.js](https://github.com/Cloud-CV/Fabrik/blob/master/ide/static/js/filterbar.js) in a text editor, add ```"your_layer_id"``` to 1(or more) of 3 framework filter array ```var KerasLayers = [...]```, ```var TensorFlowLayers = [...]``` or ```var CaffeLayers = [...]```. This should be like this ```var KerasLayers = ["RNN_Button", "GRU_Button", "your_layer_id"]```. This arrays are placed inside ```changeEvent() {}``` function.
-
-### Adding layer handling to the backend
-
-- Open [import_prototxt.py](https://github.com/Cloud-CV/Fabrik/blob/master/caffe_app/views/import_prototxt.py) file in a text editor.
-
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerImportPrototxt1.png" />
-
-- Add a function for the new layer below the category of this layer.
-- Load the parameters, do the calculations for your layer in pyhton and return the value of ```params``` (parameters).
-- Move down in the file.
-
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerImportPrototxt2.png" />
 
-- Add your defined layer in the ```layer_dict``` array, as shown above.
+- Open [ide/static/js/filterbar.js](https://github.com/Cloud-CV/Fabrik/blob/master/ide/static/js/filterbar.js) in a text editor, add ```"your_layer_id"``` to 1(or more) of 3 framework filter array ```var KerasLayers = [...]```, ```var TensorFlowLayers = [...]``` or ```var CaffeLayers = [...]```. This should be like this ```var KerasLayers = ["RNN_Button", "GRU_Button", "your_layer_id"]```. This arrays are placed inside ```changeEvent() {}``` function.
 
-- Now, open [jsonToPrototxt.py](https://github.com/Cloud-CV/Fabrik/blob/master/ide/utils/jsonToPrototxt.py) in a text editor.
 
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerJSONtoPrototxt1.png" />
+## Adding layer handling to the backend
 
-- Add an export function for training and testing of the new layer.
-- There you need to load parameters, then train & test values and at last return the trained and tested data.
-- Move down in this file as well.
+See the guides below to see how to add layer handling to the backend:
 
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerJSONtoPrototxt2.png" />
+- [Caffe](adding_new_layers_caffe.md)
+- [Keras](adding_new_layers_keras.md)
+- [Tensorflow](adding_new_layers_tensorflow.md)
 
-- Add the export function in the ```layer_map``` array.
 
-### Testing and pushing the new layer.
+## Testing and pushing the new layer.
 
 - Run the fabrik application on you local machine by following the instructions in [README](https://github.com/Cloud-CV/Fabrik/blob/master/README.md) file.
 
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layertesting.png" />
+<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layer_testing_dropout.png" />
```

Comment:
This image is not visible in markdown, can you fix this

Reply:
@Ram81 The image should be visible once this branch is merged. This is the image: https://raw.githubusercontent.com/haabhi/Fabrik/doc_add_layer/tutorials/layer_testing_dropout.png